fw/hpack.c

```diff
@@ -1269,6 +1270,13 @@ tfw_hpack_hdr_set(TfwHPack *__restrict hp, TfwHttpReq *__restrict req,
 		break;
 	case TFW_TAG_HDR_ACCEPT:
 		parser->_hdr_tag = TFW_HTTP_HDR_RAW;
+		for (i = 1, d = d_hdr->chunks; i < d_hdr->nchunks; i++) {
+			if (d[i].len == sizeof("text/html") - 1
+			    && memcmp_fast(d[i].data, "text/html",
+					   sizeof("text/html") - 1) == 0) {
+				__set_bit(TFW_HTTP_B_ACCEPT_HTML, req->flags);
+			}
+		}
```

Comment:
Basically, chunks are network chunks, i.e. we might receive `tex` and `t/html` in two separate network packets, so this matching might fail. The data is copied, but the chunks descriptors, `len` in particular, can be wrong.

It seems we need to call the HTTP parser, see how `tfw_idx_hdr_parse_if_mod_since()` is called for `TFW_TAG_HDR_IF_MODIFIED_SINCE` at the below. Since we already have `Accept` header parser in `__h2_req_parse_accept()` it might have sense to move the header value parsing out of the function and call it from `TFW_TAG_HDR_ACCEPT` and the modified `__h2_req_parse_accept()`.

Also see that we have `TFW_STR_FOR_EACH_CHUNK` for chunks iteration.